File: backend/app/routes/whatsapp.py
from fastapi import APIRouter, Request, HTTPException
from app.config import settings
from app.services import whatsapp_service

router = APIRouter()


@router.get("/webhook")
async def verificar_webhook(request: Request):
    """Verificação do webhook pela Meta."""
    params = dict(request.query_params)
    mode = params.get("hub.mode")
    token = params.get("hub.verify_token")
    challenge = params.get("hub.challenge")

    if mode == "subscribe" and token == settings.whatsapp_webhook_secret:
        return int(challenge)
    raise HTTPException(status_code=403, detail="Token inválido")


@router.post("/webhook")
async def receber_mensagem(request: Request):
    """Recebe mensagens e eventos do WhatsApp."""
    data = await request.json()
    return {"status": "recebido"}

File: worker/src/routes/whatsapp.ts
import { Hono } from "hono";
import type { Bindings } from "../index";

export const whatsappRoutes = new Hono<{ Bindings: Bindings }>();

whatsappRoutes.get("/webhook", async (c) => {
  const mode = c.req.query("hub.mode");
  const token = c.req.query("hub.verify_token");
  const challenge = c.req.query("hub.challenge");

  if (mode === "subscribe" && token === c.env.WHATSAPP_WEBHOOK_SECRET) {
    return c.text(challenge ?? "");
  }
  return c.json({ erro: "Token inválido" }, 403);
});

whatsappRoutes.post("/webhook", async (c) => {
  const data = await c.req.json();
  // Processar mensagens recebidas do WhatsApp
  return c.json({ status: "recebido" });
});
